controlplane/pkg/apis/registry/registry.proto

```diff
@@ -17,6 +18,15 @@ message NetworkServiceEndpoint {
 message NetworkService {
     string name = 1;
     string payload = 2;
+    message Match {
+        k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector source_selector = 1;
+        message Destination{
+            k8s.io.apimachinery.pkg.apis.meta.v1.LabelSelector destination_selector = 1;
+            uint32 weight = 2;
+        }
+        Destination route = 2;
```

Comment:
Should this be

repeated Destination

?

Reply:
I was not sure about this either.